### PR TITLE
Enable Operation/Resource name V2 by default in converged agent

### PR DESCRIPTION
**Pull request by IbraheemA**
### What does this PR do?

Enable updated logic for Operation/Resource name by default for converged agent customers. The v1 logic will be deprecated eventually.

If DD_APM_FEATURES or apm_features.config are specified manually and do not include enable_operation_and_resource_name_logic_v2, the new logic will be disabled.

### Describe how to test/QA your changes

Run otel-agent with no apm_features configured; verify that the new logic is enabled (e.g. by sending spans corresponding to an http server request, and verifying that the operation name is just "http.server.request" instead of a longer name prefixed with "io.opentelemetry."). Then, run otel-agent with DD_APM_FEATURES specified to exclude enable_operation_and_resource_name_logic_v2 and verify that the new logic is disabled.

### Additional Notes

Release note:

"The `enable_operation_and_resource_name_logic_v2` flag is now set to true by default in Converged Agent. This update changes how operation and resource names are derived from OTLP spans, resulting in shorter, more readable names and better alignment with OpenTelemetry specifications."

### CODE REVIEW

**Review by mx-psi — APPROVED** [review]

File: releasenotes/notes/enable-operation-name-v2-by-default-otel-agent-e62cc29df3658bdf.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+other:
+  - |
+    The `enable_operation_and_resource_name_logic_v2` flag now defaults to true in Converged Agent. This modifies the
+    logic for computing operation and resource names from OTLP spans to produce shorter, more readable names and improve
+    alignment with OpenTelemetry specifications.
```
Comment (drichards-87) [inline releasenotes/notes/enable-operation-name-v2-by-default-otel-agent-e62cc29df3658bdf.yaml:6]:
```suggestion
    The `enable_operation_and_resource_name_logic_v2` flag is now set to true by default in Converged Agent. This update changes how operation and resource names are derived from OTLP spans, resulting in shorter, more readable names and better alignment with OpenTelemetry specifications.
```

**Review by drichards-87 — APPROVED** [review]
Left a suggestion from Docs and approved the PR.

File: releasenotes/notes/enable-operation-name-v2-by-default-otel-agent-e62cc29df3658bdf.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+---
```
Comment (truthbk) [inline releasenotes/notes/enable-operation-name-v2-by-default-otel-agent-e62cc29df3658bdf.yaml:1]: Please remove the changelog entry. We do not want these changes reflected in the chagelog at this time.

**Review by truthbk — CHANGES_REQUESTED** [review]

**Review by truthbk — APPROVED** [review]

**Pull request merged**